common/djangoapps/student/views.py

```diff
@@ -1656,9 +1657,12 @@ def _do_create_account(form, custom_form=None, site=None):
         user=user,
         **{key: form.cleaned_data.get(key) for key in profile_fields}
     )
+    language = form.cleaned_data.get('language')
     extended_profile = form.cleaned_extended_profile
     if extended_profile:
         profile.meta = json.dumps(extended_profile)
+    if language:
+        LanguageProficiency.objects.create(user_profile=profile, code=language)
```

Comment:
We should move this down below the profile model save to line 1667 and wrap it in a try/except:
```
language = form.cleaned_data.get('language')
if language:
    try:
        LanguageProficiency.objects.create(user_profile=profile, code=language)
    except Exception:  # pylint: disable=broad-except
        log.exception("LanguageProficiency creation failed for user {id}.".format(id=user_id))
        raise
```